src/apps/chifra/pkg/types/types.go

```diff
@@ -6,6 +6,20 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 )
 
+type Modeler[Raw RawData] interface {
+	Model(showHidden bool, format string) Model
+	Raw() *Raw
+}
+
+type RawData interface {
+	RawReceipt
```

Comment:
I'm curious -- will we be adding additional types here as we move forward? I think so, but not sure.

Reply:
Yes, I think we will have to. Although for raw types, we may decide to simply use `map[string]string`, so that we don't have to create new type for raw data. The disadvantage of it is that we wouldn't have type checking and non-existent keys would return `var, ok := raw["non-existent"] ---> var = ""; ok = false`. So if we use raw types in the code, we should have types for them, if not - then `map[string]string` should do the job.